<?xml version="1.0" ?>
<launch>

  <arg name="cpp" default="true"/>
  <arg name="bluefox" default="false"/>
  <arg name="flea3" default="true"/>
  <arg name="openni" default="false"/>
  <arg name="image_topic" default="/pg_13344889/image_raw" />



  <!-- Desired height in cm -->
  <arg name="des_height_cm" default="2.4"/>

  <!--options: L,R,M,A -->
  <!-- Left now, as robot pours from the left, hence M,R are in jeporady -->
  <arg name="vis_side" default="-selection L"/>

  <!-- Runs the camera drivers -->
<!--   <include file="$(find queen_cam_calibration)/launch/kinect_calib.launch">
    <arg name="bluefox" value="$(arg bluefox)"/>
    <arg name="flea3" value="$(arg flea3)"/>
  </include>
 -->
  <!-- Runs the Apriltag node -->
  <include file="$(find liquid_apriltags)/launch/new_cylinder_tag_detection.launch">
    <arg name="image" value="$(arg image_topic)"/>
  </include>


  <node name='Level_Detector_Node_cpp' pkg='liquid_level_detection' type='level_detect_node' output="screen" if="$(arg cpp)">
    <param name="desired_height_cm" type="double" value="$(arg des_height_cm)"/>
    <param name="apriltag_cut_area" type="double" value="0.025"/>
    <param name="apriltag_size" type="double" value="0.019"/>
    <param name="cylinder_height" type="double" value="0.0755"/>
    <param name="x_offset_pix" type="int" value="50"/>
    <param name="y_offset_pix" type="int" value="80"/>
    <param name="debug" type="bool" value="false"/>
    <remap from="~image" to="$(arg image_topic)"/>
    <remap from="~detected_height" to="LDHeight"/>
  </node>

  <!--height for glass cylinder 0.089m tagsize: 0.017, yoffset was 40 but that was too low, area 41.6666-->
  <!--Front tag0 was 0.02m and height of 0.091 in front -->
  <!-- New plastic reciever height is 7.6cm, tagsize = 0.019cm, Area: 38.515cm**2 -->


</launch>
